src/common/settings.h

```diff
@@ -218,6 +218,8 @@ struct Values {
                                               Category::CpuDebug};
     Setting<bool> cpuopt_ignore_memory_aborts{linkage, true, "cpuopt_ignore_memory_aborts",
                                               Category::CpuDebug};
+    Setting<bool> cpuopt_multiblock_compilation{linkage, true, "cpuopt_multiblock_compilation",
+                                                Category::CpuDebug};
```

Comment:
Link this to `cpu_debug_mode` to achieve the same effect you were going for with the custom SettingsItem
```suggestion
    Setting<bool> cpuopt_multiblock_compilation{linkage,
                                                true,
                                                "cpuopt_multiblock_compilation",
                                                Category::CpuDebug,
                                                Specialization::Default,
                                                true,
                                                false,
                                                &cpu_debug_mode};
```

Reply:
Would this not be confusing from a UI point of view? And if we're doing this here, should we not do this for the rest of the cpu debug options too?

Reply:
I think it properly conveys that you can't use the CPU debug related settings when CPU debug is off.
And yes, the rest of the CPU debugging settings should be paired but since this doesn't affect Qt and isn't related to your PR, I thought it was better to leave as-is.

Reply:
(Realizing that the way I phrased it before sounds wrong)
I think it properly conveys that you can't *change* the CPU debug related settings. I suppose I could change the behavior to disable the option instead of hide it but that also wouldn't be related to this PR.